src/Compilers/Core/Portable/PEWriter/BlobBuilder.cs

```diff
@@ -0,0 +1,699 @@
+        {
+            return ImmutableArray.Create(_buffer, start, length);
+        }
+
+        internal void WriteBytes(byte value, int count)
+        {
+            int start = Advance(count);
+            fixed (byte* buffer = _buffer)
+            {
+                byte* ptr = buffer + start;
+                for (int i = 0; i < count; i++)
+                {
+                    ptr[i] = value;
+                }
+            }
+        }
+
+        internal void WriteBytes(byte[] buffer)
+        {
+            WriteBytes(buffer, 0, buffer.Length);
+        }
+
+        internal void WriteBytes(byte[] buffer, int index, int length)
+        {
+            int start = Advance(length);
+            System.Buffer.BlockCopy(buffer, index, _buffer, start, length);
+        }
+
+        internal void WriteBytes(ImmutableArray<byte> buffer)
+        {
+            WriteBytes(buffer, 0, buffer.Length);
+        }
+
+        internal void WriteBytes(ImmutableArray<byte> buffer, int index, int length)
+        {
+            int start = Advance(length);
+            buffer.CopyTo(index, _buffer, start, length);
+        }
+
+        internal void Write(Stream stream, int length)
+        {
+            int start = Advance(length);
+            int bytesRead = stream.Read(_buffer, start, length);
+            _position = start + bytesRead;
+        }
+
+        internal void PadTo(int position)
+        {
+            WriteBytes(0, position - _position);
+        }
+
+        internal void Align(int alignment)
+        {
+            int position = _position;
+            WriteBytes(0, BitArithmeticUtilities.Align(position, alignment) - position);
+        }
+
+        internal void WriteBoolean(bool value)
+        {
+            WriteByte((byte)(value ? 1 : 0));
+        }
+
+        internal void WriteByte(byte value)
+        {
+            int start = Advance(sizeof(byte));
+            _buffer[start] = value;
+        }
+
+        internal void WriteSByte(sbyte value)
+        {
+            WriteByte(unchecked((byte)value));
+        }
+
+        internal void WriteDouble(double value)
+        {
+            int start = Advance(sizeof(double));
+            fixed (byte* ptr = _buffer)
+            {
+                *((double*)(ptr + start)) = value;
+            }
+        }
+
+        internal void WriteSingle(float value)
+        {
+            int start = Advance(sizeof(float));
+            fixed (byte* ptr = _buffer)
+            {
+                *((float*)(ptr + start)) = value;
+            }
+        }
+
+        internal void WriteInt16(short value)
+        {
+            WriteUInt16(unchecked((ushort)value));
+        }
+
+        internal void WriteUInt16(ushort value)
+        {
+            int start = Advance(sizeof(ushort));
+            fixed (byte* ptr = _buffer)
+            {
+                *((ushort*)(ptr + start)) = value;
+            }
+        }
+
+        internal void WriteUInt16BE(ushort value)
```

Comment:
Yup. I can spell it out, but it is a common convention in encoding names to use LE and BE (from wikipedia: "For Internet protocols, IANA has approved "UTF-16", "UTF-16BE", and "UTF-16LE" as the names for these encodings.").